scripts/entrypoint-rpc.sh

```
@@ -30,13 +40,22 @@ trap 'kill 0 2>/dev/null || true' TERM INT
 
 if [ -z "${ONLY_START:-}" ] || [ "$ONLY_START" = "bitcoin-cache" ]; then
   echo "[entrypoint] starting bitcoin-cache on ${BITCOIN_CACHE_HOST}:${BITCOIN_CACHE_PORT}"
+  # Pass S3 credentials via environment variables (not CLI args) for security
+  # This prevents credentials from appearing in process listings and shell history
+  S3_ACCESS_KEY="${S3_ACCESS_KEY}" \
+  S3_SECRET_KEY="${S3_SECRET_KEY}" \
+  S3_ENDPOINT="${S3_ENDPOINT}" \
```

Shellcheck (0.11.0)

[warning] This assignment is only seen by the forked process.

(SC2097)